kpops/pipeline.py

```diff
@@ -83,6 +87,71 @@ def to_yaml(self) -> str:
             self.model_dump(mode="json", by_alias=True, exclude_none=True)["components"]
         )
 
+    def build_execution_graph_from(
+        self,
+        components: list[PipelineComponent],
+        reverse: bool,
+        runner: Callable[[PipelineComponent], Coroutine],
+    ) -> Awaitable:
+        sub_graph_nodes = self.__get_graph_nodes(components)
+
+        async def run_parallel_tasks(tasks: list[Coroutine]) -> None:
+            asyncio_tasks = []
+            for coroutine in tasks:
+                asyncio_tasks.append(asyncio.create_task(coroutine))
```

Comment:
be aware of garbage collection while the task is scheduled on the event loop, which could lead to bugs. it's better to store a strong reference to background tasks
https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task